source/MRViewer/MRTouchpadWin32Handler.cpp

```diff
@@ -0,0 +1,494 @@
+#ifdef _WIN32
+#include "MRTouchpadWin32Handler.h"
+
+#include <GLFW/glfw3.h>
+#define GLFW_EXPOSE_NATIVE_WIN32
+#include <GLFW/glfw3native.h>
+
+#include <spdlog/spdlog.h>
+
+//#define EVENT_DEBUG
+#ifdef EVENT_DEBUG
+#pragma warning( push )
+#pragma warning( disable: 5204 )
+#include <comdef.h>
+#pragma warning( pop )
+#endif
+
+namespace
+{
+
+class TouchpadWin32HandlerRegistry
+{
+public:
+    static TouchpadWin32HandlerRegistry& instance()
```

Comment:
why make singleton in cpp file, mb just make registry part of needed class?

Reply:
Replaced the singleton with static map.